akka-stream-tests/src/test/scala/akka/stream/scaladsl/AcknowledgeSinkSpec.scala

```diff
@@ -21,18 +21,15 @@ class AcknowledgeSinkSpec extends AkkaSpec {
 
   val noMsgTimeout = 300.millis
 
-  def assertSuccess(value: Any, fb: Future[Option[Any]]): Unit =
-    Await.result(fb, 1.second) should be(Some(value))
-
   "An AcknowledgeSink" must {
 
     "send the elements as result of future" in assertAllStagesStopped {
-      val queue = Source(List(1, 2, 3)).runWith(Sink.queue(3))
-      assertSuccess(1, queue.pull())
-      assertSuccess(2, queue.pull())
-      assertSuccess(3, queue.pull())
-      queue.pull().pipeTo(testActor)
-      expectMsg(None)
+      val expected = List(Some(1), Some(2), Some(3), None)
+      val queue = Source(expected.flatten).runWith(Sink.queue(3))
+      expected foreach { v ⇒
+        queue.pull() pipeTo testActor
+        expectMsg(v)
+      }
```

Comment:
AFAICS this does not make the test more reliable—it already expresses the right thing—but it is certainly an improvement.